src/mod_presence.erl

```diff
@@ -433,7 +433,7 @@ presence_track(Acc, FromJid, ToJid, Packet, _, <<"unsubscribe">>) ->
 presence_track(Acc, FromJid, ToJid, Packet, _, <<"unsubscribed">>) ->
     process_presence_track_subscription_and_route(Acc, FromJid, ToJid, Packet, unsubscribed);
 presence_track(Acc, FromJid, ToJid, Packet, _, <<"error">>) ->
-    ejabberd_router:route(ToJid, FromJid, Acc, Packet),
+    ejabberd_router:route(FromJid, ToJid, Acc, Packet),
     Acc;
 presence_track(Acc, FromJid, ToJid, Packet, _, <<"probe">>) ->
     ejabberd_router:route(ToJid, FromJid, Acc, Packet),
```

Comment:
https://github.com/esl/MongooseIM/blob/759b64aee06668e6130c5f7d138637f2459fc5f0/src/ejabberd_c2s.erl#L1393 because probes work that way, it's all very confusing in the old c2s code and I mixed code-paths when porting the old code, but if followed carefully, only probes are meant to invert the from-to, and when I ported I made both probes and errors invert them.